electron/src/db/index.ts

```diff
@@ -14,31 +14,38 @@ export function closeDB(db: sqlite3.Database) {
   db.close();
 }
 
-export async function checkIfRecordsExist(
+export async function getRecordCounts(
   db: sqlite3.Database,
   checkQuery: string
-): Promise<boolean> {
+): Promise<number> {
   const checkResult = await sqlite3Wrapper.allP(db, checkQuery);
   if (checkResult && Number(checkResult[0].count) > 0) {
     log.info(`${checkResult[0].count} records found`);
-    return true;
+    return checkResult[0].count;
   }
   log.info(`${db}: No records found`);
-  return false;
+  return 0;
 }
 
-export async function returnDBIfPopulated(
+export interface DBRecordCount {
+  db: sqlite3.Database;
+  recordCount: number;
+}
+
+export async function getDBWithRecordCounts(
   dbPath: string,
   checkQuery: string
-): Promise<sqlite3.Database | undefined> {
+): Promise<DBRecordCount | undefined> {
   log.info(`Attempting to initialize ${dbPath}`);
   if (fs.existsSync(dbPath)) {
     const db = initializeDB(dbPath);
-    if (await checkIfRecordsExist(db, checkQuery)) {
-      log.info(`${db} populated`);
-      return db;
+    const recordCount = await getRecordCounts(db, checkQuery);
+    if (recordCount > 0) {
+      log.info(`${dbPath} populated`);
+      return { db, recordCount };
```

Comment:
is this return redundant?

Reply:
Are you suggesting it is bc of the return on line 48?

I'll change the logic around so we dont have two returns. It's currently not redundant cause we wanna only closeDB if recordCount < 1